tests/basic_one_way_send.go

```diff
@@ -0,0 +1,99 @@
+package tests
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ava-labs/teleporter/tests/utils"
+	. "github.com/ava-labs/teleporter/tests/utils"
+
+	"github.com/ava-labs/awm-relayer/messages/teleporter"
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/subnet-evm/interfaces"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	. "github.com/onsi/gomega"
+)
+
+// Ginkgo describe node that acts as a container for the teleporter e2e tests. This test suite
+// will run through the following steps in order:
+// 1. Send a transaction to the Teleporter contract on Subnet A
+// 2. Aggregate signatures and send the Warp message to Subnet B
+// 3. Verify receipt of the message on Subnet B
+func BasicOneWaySend() {
+	var (
+		teleporterMessageID *big.Int
+	)
+
+	subnetAInfo := utils.GetSubnetATestInfo()
+	subnetBInfo := utils.GetSubnetBTestInfo()
+	teleporterContractAddress := utils.GetTeleporterContractAddress()
+	fundedAddress, fundedKey := utils.GetFundedAccountInfo()
+
+	//
+	// Send a transaction to Subnet A to issue a Warp Message from the Teleporter contract to Subnet B
+	//
+	ctx := context.Background()
+
+	nonceA, err := subnetAInfo.ChainWSClient.NonceAt(ctx, fundedAddress, nil)
+	Expect(err).Should(BeNil())
+
+	data, err := teleporter.EVMTeleporterContractABI.Pack(
+		"sendCrossChainMessage",
+		SendCrossChainMessageInput{
+			DestinationChainID: subnetBInfo.BlockchainID,
+			DestinationAddress: fundedAddress,
+			FeeInfo: FeeInfo{
+				ContractAddress: fundedAddress,
+				Amount:          big.NewInt(0),
+			},
+			RequiredGasLimit:        big.NewInt(1),
+			AllowedRelayerAddresses: []common.Address{},
+			Message:                 []byte{1, 2, 3, 4},
+		},
+	)
+	Expect(err).Should(BeNil())
+
+	// Send a transaction to the Teleporter contract
+	tx := NewTestTeleporterTransaction(subnetAInfo.ChainIDInt, teleporterContractAddress, nonceA, data)
+
+	txSigner := types.LatestSignerForChainID(subnetAInfo.ChainIDInt)
+	signedTx, err := types.SignTx(tx, txSigner, fundedKey)
+	Expect(err).Should(BeNil())
+
+	log.Info("Sending Teleporter transaction on source chain", "destinationChainID", subnetBInfo.BlockchainID, "txHash", signedTx.Hash())
```

Comment:
Would it be worthwhile to move this portion to a "SendTeleporterMessageTransaction" helper that takes the `SendCrossChainMessageInput` as param?

Reply:
I packaged this code into the helper `CreateTeleporterTransaction`, which returns the signed transaction. This is then passed to `SendTransactionAndWaitForAcceptance`.